client/core/trade.go

```diff
@@ -527,6 +527,48 @@ func (t *trackedTrade) counterPartyConfirms(match *matchTracker) (have, needed u
 	return
 }
 
+// deleteStaleCancelOrder checks if this trade has an associated cancel order,
+// and deletes the cancel order if it (i.e. the cancel order) remains at Epoch
+// status more than 15 minutes after the close of the cancel order's epoch.
+// Deleting the stale cancel order from this trade makes it possible for the
+// client to re-attempt cancelling the order.
+//
+// NOTE:
+// Stale cancel orders would be Executed if their preimage was sent or Revoked
+// if their preimages was not sent. We cannot currently tell whether the cancel
+// order's preimage was revealed, so assume that the cancel order is Executed
+// but unmatched. Consider adding a order.PreimageRevealed field to ensure that
+// the correct final status is set for the cancel order.
+//
+// This method MUST be called with the trackedTrade mutex lock held for writes.
+func (t *trackedTrade) deleteStaleCancelOrder() {
+	if t.cancel == nil {
+		return
+	}
```

Comment:
Should also check that this trade's status is still Epoch / Booked. Cancel orders are currently not deleted from Executed / Revoked or Canceled orders; and they don't need to be deleted.